core/src/test/java/com/novoda/noplayer/internal/mediaplayer/DelayedActionExecutorTest.java

```diff
@@ -0,0 +1,86 @@
+package com.novoda.noplayer.internal.mediaplayer;
+
+import android.os.Handler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.willAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class DelayedActionExecutorTest {
+
+    private static final long ANY_DELAY_IN_MILLIS = 10;
+
+    private final DelayedActionExecutor.Action action = mock(DelayedActionExecutor.Action.class);
```

Comment:
We've had this conversation on another project recently about the performance impact of using the annotation so I decided to do it here. Hopefully we will edit our tests in future to reflect this!